addons/binding/org.openhab.binding.co2signal/src/main/java/org/openhab/binding/co2signal/handler/CO2SignalHandler.java

```diff
@@ -0,0 +1,292 @@
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.co2signal.CO2SignalBindingConstants;
+import org.openhab.binding.co2signal.internal.CO2SignalConfiguration;
+import org.openhab.binding.co2signal.internal.json.CO2SignalJsonResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link CO2SignalHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jens Viebig - Initial contribution
+ */
+public class CO2SignalHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(CO2SignalHandler.class);
+
+    private static final String URL = "https://api.co2signal.com/v1/latest?%QUERY%";
+
+    private static final int DEFAULT_REFRESH_PERIOD = 30;
+
+    private ScheduledFuture<?> refreshJob;
+
+    CO2SignalJsonResponse co2Response;
+
+    private Gson gson;
+
+    public CO2SignalHandler(@NonNull Thing thing) {
+        super(thing);
+        gson = new Gson();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing CO2 Signal handler.");
+
+        CO2SignalConfiguration config = getConfigAs(CO2SignalConfiguration.class);
+        logger.debug("config apikey = (omitted from logging)");
+        logger.debug("config location = {}", config.location);
+        logger.debug("config countryCode = {}", config.countryCode);
+        logger.debug("config refresh = {}", config.refresh);
+
+        boolean validConfig = true;
+        String errorMsg = null;
+
+        if (StringUtils.trimToNull(config.apikey) == null) {
+            errorMsg = "Parameter 'apikey' is mandatory and must be configured";
+            validConfig = false;
+        }
+        if (StringUtils.trimToNull(config.location) == null && config.countryCode == null) {
+            errorMsg = "Parameter 'location' or 'stationId' is mandatory and must be configured";
+            validConfig = false;
+        }
+        if (config.refresh != null && config.refresh < 1) {
+            errorMsg = "Parameter 'refresh' must be at least 1 minute";
+            validConfig = false;
+        }
+
+        if (validConfig) {
+            updateStatus(ThingStatus.ONLINE);
+            startAutomaticRefresh();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+        }
+    }
+
+    /**
+     * Start the job refreshing the CO2 Signal data
+     */
+    private void startAutomaticRefresh() {
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            Runnable runnable = new Runnable() {
```

Comment:
Java 8 already allows for a way shorter notation of this :-)

Reply:
Please give me a hint ;-) What are you thinking about ?

Reply:
```
() -> { 
    // code
};
```

Reply:
Thanks! Fixed